tex/gregoriotex-syllable.tex

```diff
@@ -1091,13 +1091,17 @@
       \GreNoBreak%
     \fi%
     \ifgre@eolshiftsenabled%
-      %this shift will shorten the line somewhat to allow the lyrics to extend under the custos should TeX decide to insert a line break after the syllable being ended
-      \gre@calculate@eolshift{\dimexpr(\gre@dimen@enddifference-\gre@skip@bar@lastskip)\relax }%
-      \gre@debugmsg{barspacing}{End of line shift: \the\gre@dimen@eolshift}%
-      \ifdim\gre@dimen@eolshift<\gre@skip@bar@lastskip %
-        \gre@dimen@eolshift=\gre@skip@bar@lastskip\relax %
-      \fi %
-      \kern\glueexpr(-\gre@dimen@eolshift)\relax%
+      \ifdim\wd\gre@box@syllablenotes>0pt\relax%
```

Comment:
A "real element" is something that has (or could have) notes in it.  Examples of things that are not "real elements": spaces `(/)`, texverb `([ev:...])`, no-line-break area `(<nlba>)`, above lines text `([alt:...])`, clef `(c4)`, bar `(::)`, end-of-line `(Z)`.